onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -923,6 +923,22 @@ namespace OperatorHelper
         const uint32_t inputDimCount = gsl::narrow_cast<int32_t>(inputDimensions.size());
         const uint32_t axis = operatorAttributes.GetOptionalAttribute<int32_t>(AttrName::Axis, 0);
         m_axis = static_cast<int>(HandleNegativeAxis(axis, inputDimCount));
+
+        if (opsetVersion >= 18) // num_outputs attribute is only defined in opset18.
```

Comment:
🤔 Food for thought. The explicit opset check *is* sometimes needed, but generally I try to avoid adding logic complicating branches for explicit opsets and instead just check for presence of attributes (kinda like how in Windows you don't check for specific Windows versions, but rather QI for a feature existence). Since nothing bad happens when trying to query an attribute that doesn't exist, we could just delete this extra branch. 🤷‍♀️ ` if (opsetVersion >= 18)` (but I'm also okay with it being there, mainly trying to linearize the flow of the logic)